unitTests/ui/CategoryFacet/CategoryValueTest.ts

```diff
@@ -6,59 +6,107 @@ import { $$, KEYBOARD } from '../../../src/Core';
-      });
+    beforeEach(() => {
+      initCategoryValueDescriptor();
+      initCategoryFacetOptions();
+      initCategoryFacet();
+    });
+
+    it('when at maximumDepth, it does not call changeActivePath', () => {
+      const categoryValue = buildCategoryValue().makeSelectable();
+      spyOn(categoryFacet, 'changeActivePath');
+      $$($$(categoryValue.element).find('.coveo-category-facet-value-label')).trigger('click');
+      expect(categoryFacet.changeActivePath).not.toHaveBeenCalled();
+    });
 
-      it('calls changeActivePath on click when below or equal maximumDepth', () => {
-        const categoryValue = makeCategoryValue(['1', '2']).makeSelectable();
+    describe('when below maximumDepth, when it is selectable', () => {
+      beforeEach(() => (categoryValueDescriptor.path = ['1', '2']));
+
+      it('calls changeActivePath on click', () => {
+        const categoryValue = buildCategoryValue().makeSelectable();
         spyOn(categoryFacet, 'changeActivePath');
         $$($$(categoryValue.element).find('.coveo-category-facet-value-label')).trigger('click');
         expect(categoryFacet.changeActivePath).toHaveBeenCalled();
       });
 
-      it('calls changeActivePath on enter keyup when below or equal maximumDepth', () => {
-        const categoryValue = makeCategoryValue(['1', '2']).makeSelectable();
+      it('calls changeActivePath on enter keyup', () => {
+        const categoryValue = buildCategoryValue().makeSelectable();
         spyOn(categoryFacet, 'changeActivePath');
         Simulate.keyUp($$(categoryValue.element).find('.coveo-category-facet-value-label'), KEYBOARD.ENTER);
         expect(categoryFacet.changeActivePath).toHaveBeenCalled();
       });
     });
 
     it('does not call changeActivePath on click by default', () => {
-      const categoryValue = makeCategoryValue();
+      const categoryValue = buildCategoryValue();
       spyOn(categoryFacet, 'changeActivePath');
       $$($$(categoryValue.element).find('.coveo-category-facet-value-label')).trigger('click');
       expect(categoryFacet.changeActivePath).not.toHaveBeenCalled();
     });
 
     it('does not call changeActivePath on enter keyup by default', () => {
-      const categoryValue = makeCategoryValue();
+      const categoryValue = buildCategoryValue();
       spyOn(categoryFacet, 'changeActivePath');
       Simulate.keyUp($$(categoryValue.element).find('.coveo-category-facet-value-label'), KEYBOARD.ENTER);
       expect(categoryFacet.changeActivePath).not.toHaveBeenCalled();
     });
+
+    it('displays the categoryValueDescriptor #value by default', () => {
+      const categoryValue = buildCategoryValue();
+      const facetValue = categoryValue.element.find('.coveo-category-facet-value-caption').textContent;
+      expect(facetValue).toBe(categoryValueDescriptor.value);
+    });
+
+    describe(`when the categoryFacet #valueCaption option has a key that matches the categoryValueDescriptor #value`, () => {
```

Comment:
I have a feeling were not testing what's really interesting in this facet ?

We need to be testing against caption which are "one" level deep, and "multiple level" deep ?

Eg: `foo`, `foo|bar`, `foo|bar|buzz` , and some other corner case, where for example, a "key" could be a sub part of a longer path (just to make sure the code is looking for exact match).

Reply:
The look-up at the moment is 1-to-1. I think this is better than needing to specify the path as the key because 1) it is simpler, 2) it is more flexible in that you only need to add a caption once for values that appear under different paths.